documentation/managing_modules.md

```diff
@@ -0,0 +1,290 @@
+
+#### Forcibly installing modules
+When you add a module to a project, or install a project's modules, Bolt
+compares the module declarations in the project configuration file to the
+modules in the Puppetfile to check if an existing Puppetfile is managed by Bolt.
+If the Puppetfile is missing any module declarations, Bolt assumes the
+Puppetfile is not managed by Bolt and raises an error like this:
+
+```shell
+Puppetfile /myproject/Puppetfile is missing specifications for the following
+module declarations:
+
+- name: puppetlabs-ruby-task-helper
+
+This may not be a Puppetfile managed by Bolt. To forcibly overwrite the
+Puppetfile, run 'bolt module install --force'.
+```
+
+This error usually only occurs if you've manually modified your
+`bolt-project.yaml` or `Puppetfile`. To resolve the conflict and install the
+modules declared in `bolt-project.yaml`, run the following command:
+
+*\*nix shell command*
+```shell
+bolt module install --force 
+```
+
+*Powershell cmdlet*
+```shell
+Install-BoltModule -Force
+```
+
+## Pin a module version
+
+If you need to pin a module in your Bolt project to a specific version, you can
+add a version requirement to the module in your `bolt-project.yaml` and run
+`bolt module install --force`.
+
+Follow these steps to pin a module version:
+1. Find the module under the `modules` key in your `bolt-project.yaml` file.
+2. Add a `version_requirement` key that specifies the version requirements for
+   the module. For example, the following example from a `bolt-project.yaml`
+   file sets the Ruby Task Helper module to greater than or equal to 1.0.0, but
+   less than 3.0.0:
+   ```yaml
+   # bolt-project.yaml
+   ...
+   modules:
+   - name: puppetlabs/apache
+     version_requirement: '>= 4.0.0 < 6.0.0'
+   ```
+3. Run `bolt module install --force`. The `--force` flag is required because
+   you've made a change to your `bolt-project.yaml` file, and it no longer
+   matches the Puppetfile.
+
+When you run `bolt module install --force`, Bolt attempts to resolve all of the
+modules in the project configuration file to ensure that there are no version
+conflicts between modules and their dependencies. Bolt fails with an error
+message if it cannot resolve a dependency due to a version requirement. 
+
+> 🔩 **Tip**: For information on how to specify module versions, see the Puppet
+> documentation on [Specifying
+> versions](https://puppet.com/docs/puppet/latest/modules_metadata.html#specifying-versions).
+
+## Update your project's modules
+
+To update the modules in your project, run:
+
+*\*nix shell command*
+```shell
+bolt module install --force 
+```
+
+*Powershell cmdlet*
+```shell
+Install-BoltModule -Force
+```
+
+Bolt updates your modules to the latest available versions and attempts to
+resolve all of the modules in the project configuration file to ensure that
+there are no version conflicts between modules and their dependencies.
+
+## Examples
+
+### Example: Migrating an existing Bolt project
```

Comment:
```suggestion
### Migrating an existing Bolt project
```

The extra "Example" seems redundant.

Reply:
Yeah, these are kinda gross. I'm a little afraid people will navigate straight to these and get confused. But maybe I need to have more faith 😄